src/bin/upgrade/main.rs

```diff
@@ -100,45 +107,105 @@ fn update_manifest(
     Ok(())
 }
 
-/// Get a list of the paths of all the (non-virtual) manifests in the workspace.
-fn get_workspace_manifests(manifest_path: &Option<String>) -> Result<Vec<String>> {
-    let mut metadata_gatherer = Command::new("cargo");
-    metadata_gatherer.args(&["metadata", "--no-deps", "--format-version", "1", "-q"]);
+fn update_manifest_from_cache(
```

Comment:
Could you reduce duplication with ```update_manifest```?

Reply:
Indeed. That should be better.

I've also renamed from `update...` to `upgrade...`, which is arguable more accurate.